pages/enroll/enroll.ejs

```diff
@@ -15,53 +21,93 @@
         <div class="card-header bg-primary text-white">Courses</div>
         <table class="table table-sm table-hover table-striped">
           <tbody>
-            <% course_instances.forEach(function(course_instance) { %>
+            <%# var modalIdx=0; must come before the forEach: %>
```

Comment:
While we are switching to modals, we could also switch to using just one copy of each modal at the top of the page, and then call them with parameters. For an example of this, see the modals that @nwalters512  cleaned up on https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/instructorAssessmentInstances/instructorAssessmentInstances.ejs

I don’t insist on doing this, and I’m happy to merge the code as is, but I think this modal-with-parameters approach that @nwalters512 used is generally cleaner than the way I was doing it originally. It also makes the HTML much smaller, so faster page loads! And it means we don’t need the `modalIdx` variable.

Reply:
That's a good idea. The thought had crossed my mind but I wasn't sure how drastic to make the first draft of this. I can try to amend the PR a little later.

Reply:
excellent! I’ll hold of on merging this PR for now, if you plan on making that change soon. If you’d rather not make the change and want to merge as is, then just let me know.